packages/back-end/src/queryRunners/QueryRunner.ts

```diff
@@ -313,63 +451,36 @@ export abstract class QueryRunner<
             logger.debug("Reusing previous query. Already finished");
             this.onQueryFinish();
           }
-
+          logger.debug("Updating query to copy cached query");
+          const updatedDoc = await updateQuery(doc, {
+            rawResult: existing.rawResult,
+            result: existing.result,
+            status: existing.status,
+            cachedQueryUsed: existing.id,
+            statistics: existing.statistics,
+            startedAt: existing.startedAt,
+            finishedAt: existing.finishedAt,
+          });
```

Comment:
It's best if we can avoid creating and then immediately updating a doc in Mongo.  Ideally, we can just create it once with all of the properties we want.